src/app/business/block/bucket-detail/acl/acl.component.ts

```diff
@@ -37,29 +39,36 @@ export class AclComponent implements OnInit {
     //query acl list
     getAclList() {
         window['getAkSkList'](()=> {
+            
             let requestMethod = "GET";
-            let url = this.BucketService.url + '/' + this.bucketId+'/?acl';
-            window['canonicalString'](requestMethod, url, () => {
-                let options: any = {};
-                let key = "/?acl"
-                this.getSignature(options);
-                let name = this.bucketId + key
-                this.BucketService.getAcl(name,options).subscribe((res) => {
-                    let str = res['_body'];
-                    let x2js = new X2JS();
-                    let jsonObj = x2js.xml_str2json(str);
-                    let chooseObj = jsonObj.AccessControlPolicy && jsonObj.AccessControlPolicy.AccessControlList.Grant
-                    this.checkBoxArr = []
-                    Array.isArray(chooseObj) && chooseObj.forEach((item) => {
-                        if(item.Permission =="WRITE"){
-                            this.checkBoxArr.push('write','read')
-                            return
-                        }else if(item.Permission == "READ"){
-                            this.checkBoxArr.push('read')
+            let url = "/"+this.bucketId+"/?acl";
```

Comment:
Isn't there a way to keep constant files where all these URIs and other constants be kept and refrenced. Makes better code maintaainability

Reply:
The code referenced here is the URL being prepared for signature generation .This is not a constant and will require the bucketId which is a variable. 
The Constants are maintained in the Consts.ts file located under shared/utils/Consts.ts
This is already noted as an issue #84. 
This is planned to be taken as part of the requirements in the New Design for the dashboard. 
For the current codebase it will involve some effort to trace all the occurrences of various URLs and constants and then put them together.

Reply:
My bad to call it constant file..I meant a util/common function..You just pass the bucketId and get the URL...
Advantage: If the URL changes, you don't need to go in all the files and replace but just one place

Reply:
Got it. The issue #84 is infact raised for the same. Just that it was a considerable effort to change in the current dashboard and better implemented in a fresh one.

Reply:
ok